utils/check_code_style.py

```diff
@@ -49,7 +49,7 @@
 	"\\s+$": "trailing whitespace at end of line",
 	# Matches any number of operators that have no leading whitespace,
 	# except if preceded by '(', '[' or '{', or inside a 'case' constant expression.
-	"(?<!^case\\s.*)([^([{\\s" + std_op + "](?<!^.*[^\\w0-9]?operator))[" + std_op + "]+([^.\\)" + std_op + "]|$)(?!\\.\\.\\.)": "missing whitespace before operator"
+	"(?<!^case\\s.*)([^([{\\s" + std_op + "](?<!^.*[^\\w0-9]?operator))[" + std_op + "]+(?<!\\w->[*&])([^.\\)" + std_op + "]|$)(?!\\.\\.\\.)": "missing whitespace before operator"
```

Comment:
I don't think you need to support the cases:
`::&`, `.&`, or `->&`, only:
`::*`, `.*`, and `->*`.
The latter three are "pointer-to-member` operators.
I don't think the first three are valid C++ syntax. I certainly can't think of what they would mean.

Reply:
Can you give an example with `::*` and `.*`? I'm not sure we would want to format them that way.

Reply:
`.*` is basically the same as `->*` except it'd be used where the first operand is the object itself (or a reference to it), not a pointer.
`A.*p` is equivalent to `(&A)->*p`
`B->*p` is equivalent to `(*B).*p`

`::*` is used in the type of a pointer-to-member variable.
For example, we may have a class A. `A::*` would then be the type of a pointer to a member of that class.
A pointer-to-member differs from an ordinary point (that could happen to point to a member of a class) in that they are not absolute memory addresses. They can be thought of as relative to that object type.
If the class A contains a non-static member variable `thing`, you would not create a pointer-to-member for `A::thing` by taking the address of an instance of `thing`, instead you would take the address of `A::thing`, which isn't actually an instance of `thing`. Trying to dereference that pointer alone will not go well. It needs to be used in conjunction with an instance of A, using `.*` or `->*`.
The pointer-to-member can be thought of an offset for where within an object of type A the member `thing` can be found.

```c++
#include <iostream>

class A {
public:
    A() : thing(15) {}
    A(int in) : thing(in) {}
    A::* pThing() { return &A::thing; }
private:
    int thing;
};

void main()
{
    A a1(72);
    A *a2 = new A();
    A::* thingPointer = A::pThing();
    std::cout << a1.*thingPointer << '\n';
    std::cout << a2->*thingPointer << '\n';
}
```
This would (probably) print:
```
72
15
```

These pages might do a better job of explaining it than me:
https://en.cppreference.com/w/cpp/language/pointer#Pointers_to_members
https://www.ibm.com/docs/en/i/7.4?topic=only-pointers-members-c

Reply:
~~Actually, your example fails to compile.~~

This `::*` stuff seems cool.

Reply:
I'll test the fix in a bit.